app/src/main/java/org/wordpress/pioupiou/postlist/PostListActivity.java

```diff
@@ -124,17 +200,100 @@ public void onDismissed() {
         sendButton.setOnClickListener(new OnClickListener() {
             @Override
             public void onClick(View v) {
+                mNewPostContent = editText.getText().toString();
                 // Clear the saved text
                 editText.setText("");
                 tooltip.dismiss(true);
-                publishPost(editText.getText().toString());
+                publishPost();
             }
         });
     }
 
     // Publish post
-    private void publishPost(String text) {
-        Timber.i("Publishing post: " + text);
-        // TODO: Prepare the post and dispatch the action
+    private void publishPost() {
+        if (TextUtils.isEmpty(mNewPostContent)) {
+            ToastUtils.showToast(this, "Can't publish an empty post");
+            return;
+        }
+
+        // instantiate a new post, event handler will take care of setting the content and uploading it
+        PostStore.InstantiatePostPayload payload = new PostStore.InstantiatePostPayload(getSite(), false);
+        mDispatcher.dispatch(PostActionBuilder.newInstantiatePostAction(payload));
+    }
+
+    private boolean hasPostFragment() {
+        return getPostFragment() != null;
+    }
+
+    private PostListFragment getPostFragment() {
+        Fragment fragment = getFragmentManager().findFragmentById(R.id.list);
+        if (fragment instanceof PostListFragment) {
+            return (PostListFragment) fragment;
+        } else {
+            return null;
+        }
+    }
+
+    private void showProgress(boolean show) {
+        View progress = findViewById(R.id.progress);
+        progress.setVisibility(show ? View.VISIBLE : View.GONE);
+    }
+
+    private void showError(String message) {
+        ToastUtils.showToast(this, message, ToastUtils.Duration.LONG);
+    }
+
+    /*
+     * called whenever the post list changes, such as when a fetch posts has completed
+     */
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPostChanged(PostStore.OnPostChanged event) {
+        if (event.causeOfChange == PostAction.FETCH_POSTS
+                || event.causeOfChange == PostAction.FETCHED_POSTS) {
```

Comment:
`event.causeOfChange` won't ever point to 'FluxC internal-only' events like `FETCHED_POSTS` (they're marked 'Remote responses' in the [action enums](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/e013c94a64ea0b3139e9eb3f3b38a5a196d7a133/fluxc/src/main/java/org/wordpress/android/fluxc/action/PostAction.java#L27-L35) - so we can remove this line. Eventually these won't be visible at all outside the FluxC project and this will be clearer.